addons/l10n_at/data/account_fiscal_position_template.xml

```diff
@@ -4,8 +4,15 @@
         <!-- Vorlagen: Steuerliche Positionen -->
 
         <record id="fiscal_position_template_national" model="account.fiscal.position.template">
```

Comment:
@william-andre @wtaferner I do not agree.  We should keep the National position as it was as if it is national, it does not matter if it has VAT or not.  It only matters for the EU.

Reply:
Auto detection does not work for Austrian companies having a VAT, they will become the wrong fiscal position...

Reply:
@wtaferner Testing it out, it is indeed wrong the way it is now.  But if we have the vat_required mit UID on the National, then it will go to the other new EU fiscal position you added without VAT required (which we mostly call EU private).

Reply:
Indeed, the solution provided by @jco-odoo is in line with the way fiscal positions work best, and consistent between localizations. Easier to maintain and explain when consistent between similar localizations (all EU countries are similar in this regard).

Reply:
@jco-odoo 
It is about the sequence and IMO it is correctly aligned

National (ohne UID) -> limited to Austria (catch all for Austria w/o UID)
National (mit UID) -> limited to Austria (catch all for Austria w/ UID)
Europäische Union -> limited to "Europe" will catch all in the Country Group with UID
Europäische Union (ohne UID) -> limited to "Europe" will catch all those of the Country Group w/o UID (otherwise those would land in the last FP
Drittstaaten -> catch all others

We could change position of EU (w/ and w/o) but auto detection should be ok as suggested.

EU private -> it is rarely private as OSS is in place, mostly it is business not yet having a VAT number or a lack of providing it...based on this you additionally need to take care about the limits as before.

If a business tries to deduct the VAT billed through OSS it is forbidden to deduct the local VAT, this is why we need this extra fiscal position within the EU which is basically for B2B the local tax.

Reply:
assumption that fiscal positions are based on their sequence only is incorrect

Reply:
AFAIK, the FP auto-detection is a bunch code with a lot of `if` adding things to a search domain:

https://github.com/odoo/odoo/blob/5d82f678d33d20aa42ba9739a6d14cb6554be0b7/addons/account/models/partner.py#L168

I have already talked several times with Odoo accounting devs about this in Odoo XPs, but it seems it continues the same.

IMO this should be sequence-based for being able to map several conditions (with or without VAT, VIES or not, etc) to finally got the desired fiscal position, specially when you business is both B2B/B2C.

Reply:
@pedrobaeza Should we really map a tax different if VAT or VIES?  The VIES check was added for the vat_required.

Reply:
Of course a possible refactoring of the fiscal position search is not something to target on a stable version. Anyway, I just want to thank you for at least having this debate around it. I hope maybe something good comes from it for future versions.

I don't think it's so risky as you say to let the users to handle this. They are doing a similar thing in pricelists for example. Maybe the sequence field (and widget) can be under a security group option "Advance fiscal position mapping" or so.

The problems with current scheme are:

- It doesn't handle all possible situations correctly. Not sure if it has been corrected lately, but an scenario of mixed B2B/B2C e-commerce with European OSS was not mapping correctly the taxes in Spain: 
  - Intracommunity taxes for B2B VIES companies.
  - Regular taxes for B2B not VIES companies.
  - OSS taxes for B2C customers.

  And the blame history of the method can show you how broken it is due to the constant fixes, patches, reverts: https://github.com/odoo/odoo/blame/5d82f678d33d20aa42ba9739a6d14cb6554be0b7/addons/account/models/partner.py#L168

- It doesn't allow easily to add more custom criteria. Look the ugly patches for getting some of this:
  - https://github.com/OCA/account-fiscal-rule/blob/16.0/account_fiscal_position_partner_type/models/account_fiscal_position.py
  - https://github.com/OCA/account-fiscal-rule/blob/13.0/account_fiscal_position_autodetect_optional_vies/models/account_fiscal_position.py

Reply:
@wtaferner If you can just remove this fiscal position (National (ohne UID)) and squash the commits, we can merge.

Reply:
@jco-odoo 
No idea why you think this is correct as auto detection will not work for an Austrian company having a VAT (I gave evidence for the need above) -> they will get the wrong fiscal position, but why not, the users on SaaS and the rest will struggle, not me...